app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt

```diff
@@ -223,8 +263,16 @@ class EditActivity : AppCompatActivity() {
     private fun triggerDevices() {
         val readyDeviceGuids = deviceStateMap.filter { entry -> entry.value.intValue() == DeviceState.READY }.keys
         val readyDevices = deviceClientMap.filter { entry -> readyDeviceGuids.contains(entry.key) }.values
```

Comment:
```suggestion
        val readyDevices = deviceStateMap.takeIf { intValue() == DeviceState.READY }.mapNotNull { deviceClientMap[it.key] }
```

Not your code, but can be cleaned up